internal/controllers/configuration/kongupstreampolicy_utils.go

```diff
@@ -246,6 +365,13 @@ func backendRefToServiceRef(routeNamespace string, br gatewayapi.BackendRef) str
 	return buildServiceReference(namespace, string(br.Name))
 }
 
-func buildServiceReference(namespace, name string) string {
-	return fmt.Sprintf("%s/%s", namespace, name)
+func buildServiceReference(namespace, name string) serviceKey {
+	return serviceKey(fmt.Sprintf("%s/%s", namespace, name))
+}
+
+func isSupportedHTTPRouteBackendRef(br gatewayapi.BackendRef) bool {
+	groupIsCoreOrNil := br.Group == nil || *br.Group == "core"
+	kindIsServiceOrNil := br.Kind == nil || *br.Kind == "Service"
+	// We only support core Services.
+	return groupIsCoreOrNil && kindIsServiceOrNil
```

Comment:
This will return `true` for `br.Group == nil` and `br.Kind == nil` (and also for pairs like `br.Group == "core"` and `br.Kind == nil`). Is that expected?

Reply:
It makes sense reading your comment 👍 But without that comment I wouldn't be so sure about it 🙃 Do you think it might be worth adding it?

Reply:
Added to https://github.com/Kong/kubernetes-ingress-controller/pull/5444 as well. 👍